File: example-microservice-java/src/main/java/com/example/java/customer/service/CustomerService.java
package com.example.java.customer.service;

import com.example.java.customer.api.dto.CustomerCreateDto;
import com.example.java.customer.api.dto.CustomerDto;
import com.example.java.customer.api.dto.CustomerUpdateDto;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface CustomerService {

    Optional<CustomerDto> getCustomer(UUID id);

    List<CustomerDto> listCustomers();

    CustomerDto createCustomer(CustomerCreateDto customerCreateDto);

    CustomerDto updateCustomer(UUID id, CustomerUpdateDto customerUpdateDto);

    void deleteCustomer(UUID id);

}

File: example-microservice-kotlin/src/main/kotlin/com/gooddata/kotlin/customer/service/CustomerService.kt
package com.example.kotlin.customer.service

import com.example.kotlin.customer.api.dto.CustomerCreateDto
import com.example.kotlin.customer.api.dto.CustomerDto
import com.example.kotlin.customer.api.dto.CustomerUpdateDto
import java.util.*

interface CustomerService {

    fun getCustomer(id: UUID): Optional<CustomerDto>

    fun listCustomers(): List<CustomerDto>

    fun createCustomer(customerCreateDto: CustomerCreateDto): CustomerDto

    fun updateCustomer(id: UUID, customerUpdateDto: CustomerUpdateDto): CustomerDto

    fun deleteCustomer(id: UUID)

}
